### PR TITLE
Jetpack Sync: Take order type into account when performing HPOS Checksums

### PR DESCRIPTION
**Pull request by fgiannar**
The Sync `WooCommerce_HPOS_Orders` module only allows syncing orders of type: `shop_order` and `shop_order_refund`.
The corresponding HPOS Checksum logic though does not respect this.
This means that while an order with type different than the allowed ones, eg `shop_subscription` is not synced, when the corresponding checksum runs on `wp_wc_orders` it will take it into account causing discrepancies. Same applies for the rest HPOS Checksums, aka: `wc_order_addresses` and `wc_order_operational_data`.
The current PR fixes this.

## Proposed changes:

* `Automattic\Jetpack\Sync\Modules\WooCommerce_HPOS_Orders`: Update `get_order_types_to_sync` method's visibility to public and make it a static one so it can be used in the corresponding HPOS Checksum logic (both on remote sites and WPCOM)
* `Automattic\Jetpack\Sync\Replicastore\Table_Checksum`: Update the HPOS Checksum configuration by limiting `wc_orders`, `wc_order_addresses` and `wc_order_operational_data` queries to the ones with order (or associated order) type: `shop_order` and `shop_order_refund`. 

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

p9dueE-8ec-p2#comment-10522

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:

D157400-code

### CODE REVIEW

**Review by darssen — COMMENTED** [review]
Many thanks for working on this.

Code looks good an tests well. Let's wait for confirmation on the needed types before merging since we might need to add some

**Comment by fgiannar** [comment]
> Code looks good an tests well. Let's wait for confirmation on the needed types before merging since we might need to add some

I'd probably handle this in a separate PR but since I'm about to go AFK for a month feel free to proceed as you think it's best :)

**Comment by fgiannar** [comment]
Unless of course we are asked to allow all types, which in that case, makes this PR redundant.

**Review by darssen — APPROVED** [review]
Tests well. As suggested here, if more types need to be included we will treat them in a separate PR. This way we can get the changes before code freeze.

**Pull request merged**
